src/WebJobs.Extensions.DurableTask.Analyzers/Analyzers/ActivityFunction/FunctionAnalyzer.cs

```diff
@@ -56,115 +57,181 @@ private void RegisterAnalyzers(CompilationAnalysisContext context)
+                });
+            }
+        }
+
+        private bool TryGetCallActivityInvocation(SyntaxNodeAnalysisContext context, out InvocationExpressionSyntax invocationExpression)
+        {
+            invocationExpression = context.Node as InvocationExpressionSyntax;
+            if (invocationExpression != null)
+            {
                 var expression = invocationExpression.Expression as MemberAccessExpressionSyntax;
                 if (expression != null)
                 {
                     var name = expression.Name;
                     if (name.ToString().StartsWith("CallActivityAsync") || name.ToString().StartsWith("CallActivityWithRetryAsync"))
                     {
-                        var functionName = invocationExpression.ArgumentList.Arguments.FirstOrDefault();
-                        var argumentType = invocationExpression.ArgumentList.Arguments.Last();
-                        var returnType = invocationExpression.ChildNodes().Where(x => x.IsKind(SyntaxKind.SimpleMemberAccessExpression))
-                            .FirstOrDefault()?
-                            .ChildNodes()
-                            .Where(x => x.IsKind(SyntaxKind.GenericName))
-                            .FirstOrDefault()?
-                            .ChildNodes()
-                            .Where(x => x.IsKind(SyntaxKind.TypeArgumentList))?
-                            .FirstOrDefault();
-                        var returnTypeName = "System.Threading.Tasks.Task";
-                        if (returnType != null)
-                        {
-                            returnTypeName = GetQualifiedTypeName(context.SemanticModel.GetTypeInfo(returnType.ChildNodes().FirstOrDefault()).Type);
-                            returnTypeName = "System.Threading.Tasks.Task<" + returnTypeName + ">";
-
-                        }
-                        var typeInfo = context.SemanticModel.GetTypeInfo(argumentType.ChildNodes().First());
-                        var typeName = "";
-                        if (typeInfo.Type == null)
-                            return;
-                        typeName = GetQualifiedTypeName(typeInfo.Type);
-                        if (functionName != null && functionName.ToString().StartsWith("\""))
-                            calledFunctions.Add(new ActivityFunctionCall
-                            {
-                                Name = functionName.ToString().Trim('"'),
-                                NameNode = functionName,
-                                ParameterNode = argumentType,
-                                ParameterType = typeName,
-                                ExpectedReturnType = returnTypeName,
-                                ExpectedReturnTypeNode = invocationExpression
-                            });
+                        return true;
                     }
                 }
             }
+
+            invocationExpression = null;
```

Comment:
This assignment is unnecessary, as it is already null at this point. In fact, you could replace the if-block with `return invocationExpresion != null`.